FreeRTOS/FreeRTOS
# add the ability to profile Posix applications with gprof

add the ability to profile Posix applications with gprof

Description
-----------
A user who ported his application to FreeRTOS Posix can now profile his application with gprof
they can also run their application with sanitizers enabled

Test Steps
-----------
```
make PROFILE=1
```
after running the application a gmon.out file will be generated.
the file can be converted with
```
make profile
```
in order to generate prof_flat.txt and prof_call_graph.txt


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
